shared/src/conversions.rs

```diff
@@ -18,6 +26,17 @@ pub fn u256_to_big_rational(input: &U256) -> BigRational {
     BigRational::new(as_bigint, 1.into())
 }
 
+pub fn big_int_to_u256(input: &BigInt) -> Result<U256> {
+    let (sign, bytes) = input.to_bytes_be();
+    if sign == Sign::Minus {
```

Comment:
nanonit: I think both these `if` statements can be replaced with `ensure` as well.